src/training/stylegan.py

```diff
@@ -0,0 +1,100 @@
+import torch
+from src.logger.simple import Logger
+from src.storage.simple import Storage
+from src.losses import ocr
+from torch import nn, optim
+from torch.utils.data import DataLoader
+from loguru import logger
+from torchvision import models
+class Trainer:
+    def __init__(self,
+                 model: nn.Module,
+                 optimizer: optim.Optimizer,
+                 scheduler: optim.lr_scheduler._LRScheduler,
+                 train_dataloader: DataLoader,
+                 val_dataloader: DataLoader,
+                 storage: Storage,
+                 logger: Logger,
+                 total_epochs: int,
+                 device: str,
+                 coef_ocr_loss: float,
+                 coef_perceptual_loss: float,
+                 perceptual_loss: nn.Module):
+        
+        self.device = device
+        self.model = model
+        self.optimizer = optimizer
+        self.scheduler = scheduler
+        self.train_dataloader = train_dataloader
+        self.val_dataloader = val_dataloader
+        self.total_epochs = total_epochs
+        self.logger = logger
+        self.storage = storage
+        self.ocr_loss = ocr.OCRLoss().to(device)
```

Comment:
Maybe I wrong, but I suppose all initialisation should happen in config and only result passed to the trainer class.